src/transformState.js

```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const elem of actions) {
+    if (elem.type === 'addProperties') {
+      Object.assign(state, elem.extraData);
+    }
+
+    if (elem.type === 'removeProperties') {
+      for (const key of elem.keysToRemove) {
+        delete state[key];
+      }
+    }
+
+    if (elem.type === 'clear') {
+      Object.keys(state).forEach(key => delete state[key]);
+    }
+  }
```

Comment:
follow the checklist
https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md